modules/EventEmitter.js

```diff
@@ -0,0 +1,139 @@
+class EventEmitter {
+  listeners = {};  // key-value pair
+
+  _newEvent(eventName, fn) {
+    if (!this.listeners[`${eventName}`]) {
+      this.listeners[`${eventName}`] = [];
+    }
+    this.listeners[`${eventName}`].push({function: fn, once: false })
+  }
+
+  _deleteEvent(eventName, fn) {
+    const arrEvents = this.listeners[`${eventName}`];
+    for (let i = 0; i < arrEvents.length; i++) {
+      if (arrEvents[i].function === fn) {
+        arrEvents.splice(i, 1);
+      }
+    }
+  }
+
+  addListener(eventName, fn) {
+    this._newEvent(eventName, fn)
+  }
+
+  on(eventName, fn) {
+    this._newEvent(eventName, fn)
+  }
+
+  removeListener(eventName, fn) {
+    this._deleteEvent(eventName, fn);
+  }
+
+  off(eventName, fn) { 
+    this._deleteEvent(eventName, fn);
+  }
+
+  once(eventName, fn) {
+    if (!this.listeners[`${eventName}`]) {
+      this.listeners[`${eventName}`] = [];
+    }
+    this.listeners[`${eventName}`].push({function: fn, once: true})
+  }
+
+  emit(eventName, ...rest) { 
+    if (this.listeners[`${eventName}`]) {
+      this.listeners[`${eventName}`].forEach(event => {
+        if (event.once) {
+          event.function(...rest)
+          const index = this.listeners[`${eventName}`].indexOf(event);
+          this.listeners[`${eventName}`].splice(index, 1);
+        } else {
+          event.function(...rest);
+        }
+      });
+    }
+    
+    
+  }
+
+  listenerCount(eventName) {
+    return this.listeners[`${eventName}`].length;
+  }
+
+  rawListeners(eventName) { 
+    return this.listeners[`${eventName}`].map(function (item, index, array) {
+      return item.function;
```

Comment:
Good, just a small suggestion, you can rewrite a bit shorter
```suggestion
    return this.listeners[`${eventName}`].map(item => item.function);
```